### ISSUE TITLE
Editorconfig: dotnet_sort_system_directives_first not obeyed on Format Document 

### ISSUE THREAD

**Issue by nevinvalsaraj** [dotnet/roslyn#38504]
**Version Used**: VS Enterprise 2019, Version 16.2.3

**Steps to Reproduce**:

1. Add `.editorconfig` file with the following setting
```
# Dotnet code style settings:
[*.{cs,vb}]
# Sort using and Import directives with System.* appearing first
dotnet_sort_system_directives_first = true
```
2. Open a .cs file with multiple `using` directives with a System directive at the bottom.
3. Format Document (Ctrl+K, Ctrl+D)

**Expected Behavior**:
'Using' directives are re-ordered with System directives moved to top.

**Actual Behavior**:
No change in order of 'using' directives.

Additional Information:

Option to specify order of 'using' directives are found both in Tools>Options as well as editorconfig. 
- Run Default Code Cleanup command (Ctrl+K, Ctrl+E) performs the re-ordering correctly when specified in .editorconfig (Value specified in .editorconfig supersedes value specified in Tool>Options, as expected).
- Format Document doesn't seem to re-order the directives at all! (Regardless of what is specified in .editorconfig or Tools>Options.

Notes on related issues: 
- #17658 - Identical issue raised for VS 2017 closed as not-reproducible.
- #22465 - the issue was raised for the `false` option and was apparently fixed in v15.3, but I'm using v16.2.3 and having issues with `true` option.

**Comment by sharwell** [comment]
@nevinvalsaraj The Format Document operation only adjusts whitespace within the document. It will not reorder items or otherwise make changes to match the current .editorconfig or Tools>Options settings. You can use the Code Cleanup operation to perform these more advanced code style cleanup operations, or use the light bulb to fix the directives in the current file, project, or solution.

/cc @JoeRobich

**Comment by nevinvalsaraj** [comment]
Thank you for the succint clarification @sharwell ! However, shouldn't any violation of .editorconfig rules be highlighted? I accept and respect the design choice that VS will not make the edits for me, but when I configure a rule in a supported format I expect a red-underline-highlight/ lightbulb refactor option.

Also, I believe this difference is left unclear [in the documentation](https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2017#add-and-remove-editorconfig-files) :

> When you format the document (Edit > Advanced > Format Document or Ctrl+K, Ctrl+D), the settings in the EditorConfig file are applied to existing lines of code.

Might be a good idea to clarify the differences there as well.

**Comment by AArnott** [comment]
This doesn't work for me even for the Remove and Sort Usings command. C# always honors the IDE setting and seems to ignore the .editorconfig file setting.

**Comment by sharwell** [comment]
@AArnott do you have a repro case for this? I have the IDE mismatched relative to .editorconfig, and it's been using the .editorconfig value for me.

**Comment by AArnott** [comment]
Yes. It repro'd for me consistently this morning in the several times I tried.

**Comment by AArnott** [comment]
https://developercommunity.visualstudio.com/content/problem/752106/editorconfig-dotnet-sort-system-directives-first-s.html

**Comment by sharwell** [comment]
📝 The issue observed by @AArnott turned out to be an edge case where a customized project configuration disabled the IDE's support for .editorconfig files. I'm going to mark those comments as resolved since we're dealing with that issue internally.

**Comment by techie55** [comment]
Hello, we are facing the same issue with **Code Cleanup** extension in our project which on Visual Studio 2019 16.3.1 on .NET Framework 4.7.1 
The weird thing is that the usings do not get sorted only for Web Projects using .NET Framework
Our other class library projects which are on .NET Framework 4.7.1 sorts the usings correctly with System directives being sorted first. 
Also our web project on dot net core also works well with **Code Cleanup** extension

To Summarize:

.NET Framework Web Project  :  usings **DO NOT** sort correctly
.NET Framework Class Library  :  usings sort correctly
.NET Core Web Project  :  usings  sort correctly

**Comment by jasonmalinowski** [comment]
@techie55: could you file a new bug for your issue? The same symptoms can be caused by a number of underlying issues and I want to make sure we don't confuse yours and the original filer's.

**Comment by sharwell** [comment]
Assigning to @kendrahavens to update the documentation per https://github.com/dotnet/roslyn/issues/38504#issuecomment-528766550

**Comment by kendrahavens** [comment]
@sharwell The docs update was completed in https://github.com/MicrosoftDocs/visualstudio-docs-pr/pull/5423

**Comment by Romanx** [comment]
Hello,

I ran into this again today using `VS 2019 Version 16.7.6`. Created an isloated repo without any local settings in a fresh project with the following two files:
```editorconfig
# .editorconfig
###############################
# .NET Coding Conventions     #
###############################
[*.{cs,vb}]
# Organize usings
dotnet_sort_system_directives_first = true
```

```
// Program.cs
using Other;
using System;

namespace ConsoleApp
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine(new Class().Foo);
        }
    }
}

namespace Other
{
    public class Class
    {
        public string Foo => "Bar";
    }
}
```

When running Code cleanup `Ctrl+K, Ctrl+E` with default profile of:
- Sort Usings
- Remove unnecessary usings

I would expect this to re-order the namespaces to have Other underneath System. As a side but possibly related note as per [this doc on formatting rules](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/formatting-rules) adding `dotnet_diagnostic.IDE0055.severity = error` doesn't seem to mark this as being a problem either.

**Comment by CyrusNajmabadi** [comment]
Format-document only touches whitespace.  Code cleanup is used to affect more than just whitespace differences.

**Comment by Romanx** [comment]
Hey @CyrusNajmabadi, code cleanup didn't have the desired effect either. I'm not sure if i was doing anything obviously wrong however

**Comment by Youssef1313** [comment]
Should this be closed per https://github.com/dotnet/roslyn/issues/38504#issuecomment-547190261?

**Comment by sharwell** [comment]
Resolving as By Design
